cpp/tests/strings/split_tests.cpp

```diff
@@ -339,6 +305,127 @@ TEST_F(StringsSplitTest, SplitRecordWhitespaceWithMaxSplit)
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
 }
 
+TEST_F(StringsSplitTest, SplitRegex)
+{
+  std::vector<const char*> h_strings{" Héllo thesé", nullptr, "are some  ", "tést String", ""};
+  auto validity =
+    thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; });
+  cudf::test::strings_column_wrapper input(h_strings.begin(), h_strings.end(), validity);
+  auto sv = cudf::strings_column_view(input);
+
+  {
+    auto result = cudf::strings::split_re(sv, "\\s+");
+
+    cudf::test::strings_column_wrapper col0({"", "", "are", "tést", ""}, validity);
+    cudf::test::strings_column_wrapper col1({"Héllo", "", "some", "String", ""}, {1, 0, 1, 1, 0});
+    cudf::test::strings_column_wrapper col2({"thesé", "", "", "", ""}, {1, 0, 1, 0, 0});
+    auto expected = cudf::table_view({col0, col1, col2});
+    CUDF_TEST_EXPECT_TABLES_EQUIVALENT(result->view(), expected);
+
+    // rsplit == split when using default parameters
+    result = cudf::strings::rsplit_re(sv, "\\s+");
+    CUDF_TEST_EXPECT_TABLES_EQUIVALENT(result->view(), expected);
+  }
+
+  {
+    auto result = cudf::strings::split_re(sv, "[eé]");
+
+    cudf::test::strings_column_wrapper col0({" H", "", "ar", "t", ""}, validity);
+    cudf::test::strings_column_wrapper col1({"llo th", "", " som", "st String", ""},
+                                            {1, 0, 1, 1, 0});
+    cudf::test::strings_column_wrapper col2({"s", "", "  ", "", ""}, {1, 0, 1, 0, 0});
+    cudf::test::strings_column_wrapper col3({"", "", "", "", ""}, {1, 0, 0, 0, 0});
+    auto expected = cudf::table_view({col0, col1, col2, col3});
+    CUDF_TEST_EXPECT_TABLES_EQUIVALENT(result->view(), expected);
+
+    // rsplit == split when using default parameters
+    result = cudf::strings::rsplit_re(sv, "[eé]");
+    CUDF_TEST_EXPECT_TABLES_EQUIVALENT(result->view(), expected);
+  }
+}
+
+TEST_F(StringsSplitTest, SplitRecordRegex)
+{
+  std::vector<const char*> h_strings{" Héllo thesé", nullptr, "are some  ", "tést String", ""};
+  auto validity =
+    thrust::make_transform_iterator(h_strings.begin(), [](auto str) { return str != nullptr; });
+  cudf::test::strings_column_wrapper input(h_strings.begin(), h_strings.end(), validity);
+  auto sv = cudf::strings_column_view(input);
+
+  using LCW = cudf::test::lists_column_wrapper<cudf::string_view>;
+  {
+    auto result = cudf::strings::split_record_re(sv, "\\s+");
```

Comment:
Can you add a test with `"\\."` pattern please? We need to verify the result for that case in Spark plugin side. Thanks.

Reply:
Can you provide some sample strings that you want to split?
Also, if you are trying to split on a literal '.' character, you should use the regular `split` and not the regex ones.

Reply:
Please check out the comment here: https://github.com/NVIDIA/spark-rapids/pull/4681/#discussion_r797928342

Reply:
Sorry, I'm not finding any example strings in the link provided. Maybe I'm just unable to see them.
You are welcome to pull this PR to try it out with the Spark testcases. 
I don't think we need a `\\.` testcase here since it would not be recommended to use regex for a single character.

Reply:
I copy the example from there to here:

> For example if I did a split with "\\." as the separator on "a.b.c" I would expect an output of ["a", "b", "c"]. But I cannot take "\\." and pass it into a non-regexp string split, because it will try to look for a "\\" character in front of the '.'. Am I wrong/missing something? I think we could still support it, but we would have to transpile the regular expression into a constant string.